static/app/components/acl/access.tsx

```diff
@@ -14,103 +15,75 @@ const DEFAULT_NO_ACCESS_MESSAGE = (
 };
 
-type ChildFunction = (props: ChildRenderProps) => React.ReactNode;
+type ChildFunction = (props: ChildRenderProps) => JSX.Element;
 
-type DefaultProps = {
+type Props = {
   /**
    * List of required access levels
    */
-  access: Scope[];
-
+  access?: Scope[];
   /**
-   * Custom renderer function for "no access" message OR `true` to use
-   * default message. `false` will suppress message.
+   * Children can be a node or a function as child.
    */
-  renderNoAccessMessage: ChildFunction | boolean;
-
+  children?: React.ReactNode | ChildFunction;
   /**
    * Requires superuser
    */
   isSuperuser?: boolean;
-
+  /**
+   * Custom renderer function for "no access" message OR `true` to use
+   * default message. `false` will suppress message.
+   */
+  renderNoAccessMessage?: ChildFunction | boolean;
   /**
    * Should the component require all access levels or just one or more.
    */
   requireAll?: boolean;
 };
 
-const defaultProps: DefaultProps = {
-  renderNoAccessMessage: false,
-  isSuperuser: false,
-  requireAll: true,
-  access: [],
-};
-
-type Props = {
-  /**
-   * Configuration from ConfigStore
-   */
-  config: Config;
-
-  /**
-   * Current Organization
-   */
-  organization: Organization;
-
-  /**
-   * Children can be a node or a function as child.
-   */
-  children?: React.ReactNode | ChildFunction;
-} & Partial<DefaultProps>;
-
 /**
  * Component to handle access restrictions.
  */
-class Access extends Component<Props> {
-  static defaultProps = defaultProps;
-
-  render() {
-    const {
-      organization,
-      config,
-      access,
-      requireAll,
-      isSuperuser,
-      renderNoAccessMessage,
-      children,
-    } = this.props;
-
-    const {access: orgAccess} = organization || {access: []};
-    const method = requireAll ? 'every' : 'some';
-
-    const hasAccess = !access || access[method](acc => orgAccess.includes(acc));
-    const hasSuperuser = !!(config.user && config.user.isSuperuser);
-
-    const renderProps: ChildRenderProps = {
-      hasAccess,
-      hasSuperuser,
-    };
-
-    const render = hasAccess && (!isSuperuser || hasSuperuser);
-
-    if (!render && typeof renderNoAccessMessage === 'function') {
-      return renderNoAccessMessage(renderProps);
-    }
-    if (!render && renderNoAccessMessage) {
-      return DEFAULT_NO_ACCESS_MESSAGE;
-    }
-
-    if (isRenderFunc<ChildFunction>(children)) {
-      return children(renderProps);
-    }
+function Access({
+  children,
+  renderNoAccessMessage = false,
+  isSuperuser = false,
+  requireAll = true,
+  access = [],
+}: Props) {
+  const config = useLegacyStore(ConfigStore);
+  const organization = useOrganization();
+
+  const {access: orgAccess} = organization || {access: []};
+  const method = requireAll ? 'every' : 'some';
+
+  const hasAccess = !access || access[method](acc => orgAccess.includes(acc));
+  const hasSuperuser = !!(config.user && config.user.isSuperuser);
+
+  const renderProps: ChildRenderProps = {
```

Comment:
Since we pass these down to the next component, it might make sense to memo and save some possible rerenders down the line.